### ISSUE TITLE
Discussion on attention mask handling

### ISSUE THREAD

**Issue by vidyasiv** [huggingface/optimum-habana#692]
### System Info

```shell
N/A
```

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Code snippet: https://github.com/huggingface/optimum-habana/pull/425/files#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eR215-R223 

### Expected behavior

I didn't see a place for discussions, so opening an issue. 
The question I had was the reasoning for adding these lines: 
- https://github.com/huggingface/optimum-habana/pull/425/files#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eR215-R223 . 
- If we compare with transformers repository https://github.com/huggingface/transformers/blob/v4.34.1/src/transformers/generation/utils.py#L787-L801 , for the encoder_decoder case only `decoder_attention_mask` is handled and not `attention_mask`.  Maybe this needs to be upstreamed or is handled elsewhere in transformers?
 
Could someone please clarify this- @bhargaveede or @regisss  ?

**Comment by vidyasiv** [comment]
> Good question! Have you tried running generation with T5 after removing this code block from the source code? Maybe @bhargaveede has more insight into this.

Hi @regisss , thanks for taking a look. Yes, currently doing some experiments with BART and T5 to verify this.

**Comment by vidyasiv** [comment]
Based on few experiments we did per your suggestion and cases per Habana recommendation, it appears that we do not need those lines (no functional errors/performance degradation).  Let us know if there's other ways to verify.

[BART inference](https://github.com/huggingface/optimum-habana/assets/81709031/b258b460-07ba-4e8e-9e9c-8d94c812b092)
[T5 Training](https://github.com/huggingface/optimum-habana/assets/81709031/11e8cb63-b4db-4be1-b68a-f9e273f77f3c)

Please ignore missing case B (Remove lines 221-224, 77 and 79), we found it was not functional with just these changes

This discovery was part of a debug. Removing these lines (cases C and D) enables quite a few translation models that were failing previously. Will be starting a PR for further review.
